### PR TITLE
NIFI-1657 - Configure travis-ci to run mvn verify in jp and fr

### PR DESCRIPTION
**Pull request by trixpan**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [X] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [X] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [X] Is your initial contribution a single, squashed commit?

### For code changes:
- [X] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [X] ~~Have you written or updated unit tests to verify your changes?~~
- [X] ~~If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?~~
- [X] ~~If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?~~
- [X] ~~If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?~~
- [X] ~~If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?~~

### For documentation related changes:
- [X] ~~Have you ensured that format looks appropriate for the output in which it is rendered?~~

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by pvillard31** [comment]
Thanks @trixpan, this will be useful!
What do you think: should we get this merged now or wait to fix the tests known to fail with specific locales? (just to avoid a lot of build failures in the meantime)

**Comment by trixpan** [comment]
@pvillard31 I changed the commit a bit so to use envs (I think things should move faster this way) I reckon we should merge and use it to correct residual issues. This second attempt should hopefully create two independent builds (env in travis lingo) so we will still be able to figure out if the issue is widespread or related to locale.

what do you think?

File: .travis.yml [hunk 1]
```diff
@@ -15,6 +15,11 @@
 
 language: java
 
+env:
+  - USER_LANGUAGE=en USER_COUNTRY=US
+  - USER_LANGUAGE=pt USER_COUNTRY=BR
+  - USER_LANGUAGE=jp USER_COUNTRY=JP
```
Comment (pvillard31) [inline .travis.yml]: ``USER_LANGUAGE=ja USER_COUNTRY=JP``
no?

**Comment by trixpan** [comment]
@pvillard31 good catch!

**Comment by trixpan** [comment]
by the way, just in case someone ever wonder the rationale behind the countries: basic English (Java's default env), a Latin and a "Asian" language with no official status at the UN.

**Comment by pvillard31** [comment]
I believe we need one more... I know what you are going to say: French people... what a pain ;)

````
123 456   fr_FR
345 987,246   fr_FR

123.456   pt_BR
345.987,246   pt_BR

123,456   ja_JP
345,987.246   ja_JP

123,456   en_US
345,987.246   en_US
````

**Comment by trixpan** [comment]
Only if you commit to port NiFi to minitel. :rofl: ...

Just kidding, done

**Comment by trixpan** [comment]
@pvillard31 seems good to go. failures were caused by TCP processors

**Comment by pvillard31** [comment]
@trixpan I just wanted to check before getting this merged but I believe setting the properties like you did does not change the locale settings for surefire plugin (unit tests). The fact is I know there are some tests failing with French locale although the travis build looks OK.

There is at least one in:
````
nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-cluster/
````
This is tracked with NIFI-2683.

So I just did some tests:

If I run
````
mvn clean install -Duser.language=fr -Duser.region=FR
````
In the above directory, the build is fine.

But if I change the parent pom.xml with what I suggested in the JIRA:
````
                <plugin>
                    <groupId>org.apache.maven.plugins</groupId>
                    <artifactId>maven-surefire-plugin</artifactId>
                    <version>2.18</version>
                    <configuration>
                        <includes>
                            <include>**/*Test.class</include>
                            <include>**/Test*.class</include>
                            <include>**/*Spec.class</include>
                        </includes>
                        <redirectTestOutputToFile>true</redirectTestOutputToFile>
                        <argLine combine.children="append">-Xmx1G -Djava.net.preferIPv4Stack=true -Duser.language=fr -Duser.region=FR</argLine>
                    </configuration>
                    <dependencies>
                        <dependency>
                            <!-- Force surefire to use JUnit -->
                            <groupId>org.apache.maven.surefire</groupId>
                            <artifactId>surefire-junit4</artifactId>
                            <version>2.18</version>
                        </dependency>
                    </dependencies>
                </plugin>
````

and run
````
mvn clean install
````

then I have a test failure:
````
Failed tests:
  StandardHttpResponseMapperSpec.MergeResponses: #responseEntities.size() HTTP 200 #httpMethod responses for #requestUriPart:122 Condition not satisfied:
````

IIRC the arguments passed along the maven command are not used by surefire plugin which uses its own environment to run unit tests.

So here is my proposition, in the parent pom, add an empty property like this:
````
    <properties>
        <maven.compiler.source>1.8</maven.compiler.source>
        <maven.compiler.target>1.8</maven.compiler.target>
        <maven.min-version>3.1.0</maven.min-version>
        <maven.surefire.arguments/>
        ...
    </properties>
````

Then, in the plugin configuration, change the argLine with:
````
<argLine combine.children="append">-Xmx1G -Djava.net.preferIPv4Stack=true ${maven.surefire.arguments}</argLine>
````

This way, there is absolutely no change with the current behavior and the default locale of the system will remain the one used when building. However you can now specify the locale to use with:
````
mvn clean install -Dmaven.surefire.arguments="-Duser.language=fr -Duser.region=FR"
````

And I confirm that the test is failing as "expected".

What do you think?

**Comment by trixpan** [comment]
seems great. will do it and force push soon

**Comment by trixpan** [comment]
by the way, good catch, I had faced this issue before but totally forgot about it.

**Comment by trixpan** [comment]
@pvillard31 - this seems to be working now

**Comment by trixpan** [comment]
@pvillard31 - after a few hundred builds and options build seems to be having as expected:

https://travis-ci.org/trixpan/nifi/builds/200568234

when environment is set to en_US or "default", build succeeds while the french build fails as expected.

https://travis-ci.org/trixpan/nifi/jobs/200568237#L2899

**Comment by pvillard31** [comment]
👍
Thanks a lot @trixpan, I know it has been a real nightmare to get things working with Travis... Will merge ;)

**Pull request closed without merging**